### DISCUSSION TITLE
Switch documentation system

### DISCUSSION THREAD

**Discussion by xf-** [rspamd/rspamd#3820] [Ideas]
First:
**THX for great software!**

Using the docu is really annoying, no search, feature like highlighting are missing.
I would recommend docusaurus
- Open source
- Modern system, but functions also without JS
- easily self-hosted and deployed with GitHub actions or some other tool
- Also based on Markdown
- Plugin lunr is a self-hosted search - index is build in deployment
- (Supports Versions, Languages .... - optional)
- Syntax highlighting

Bonus
- Dark and light Theme

Website
https://docusaurus.io/
(uses angolia and ant not lunr plugin)

I didn't encounter a missing feature.

**Comment by vstakhov** [comment]
It looks interesting but from what I see it would require massive changes to the site. I have neither time nor desire to invest my time in this topic again (well, and I don't have that time tbh).

**Reply by xf-** [reply]
Would you think about it, if you got help with it?

![image](https://user-images.githubusercontent.com/1295633/126492204-a625c960-992e-492c-a068-4e969006e11e.png)

Menu is set on autogeneration ()
![image](https://user-images.githubusercontent.com/1295633/126492362-e03b1186-182b-42b5-8e6d-57f3de1fcb14.png)

![image](https://user-images.githubusercontent.com/1295633/126492445-4b10617f-ec1d-4f53-a0be-114106f77e72.png)

**Comment by vstakhov** [comment]
Ok, let's first try to summarize what's wrong with the current system? Lack of search? Aside that, I don't see any improvements tbh.
